besu/src/main/java/org/hyperledger/besu/controller/BesuController.java

```diff
@@ -170,59 +169,44 @@ public PluginServiceFactory getAdditionalPluginServices() {
   }
 
   public static class Builder {
-
-    public BesuControllerBuilder fromEthNetworkConfig(final EthNetworkConfig ethNetworkConfig) {
-      return fromEthNetworkConfig(ethNetworkConfig, Collections.emptyMap());
-    }
-
-    public BesuControllerBuilder fromEthNetworkConfig(
-        final EthNetworkConfig ethNetworkConfig, final Map<String, String> genesisConfigOverrides) {
-      return fromGenesisConfig(
-              GenesisConfigFile.fromConfig(ethNetworkConfig.getGenesisConfig()),
-              genesisConfigOverrides)
-          .networkId(ethNetworkConfig.getNetworkId());
-    }
-
-    public BesuControllerBuilder fromGenesisConfig(final GenesisConfigFile genesisConfig) {
-      return fromGenesisConfig(genesisConfig, Collections.emptyMap());
+    public BesuControllerBuilder fromConfigOptions(
+        final GenesisConfigOptions configOptions, final EthNetworkConfig ignoredNetworkConfig) {
+      // wrap with TransitionBesuControllerBuilder if we have a terminal total difficulty:
+      final BesuControllerBuilder builder = internalFromConfigOptions(configOptions);
+      if (configOptions.getTerminalTotalDifficulty().isPresent()) {
+        if (configOptions.getTerminalTotalDifficulty().get().equals(UInt256.ZERO)) {
+          return new MergeBesuControllerBuilder();
+        } else {
+          // TODO this should be changed to vanilla MergeBesuControllerBuilder and the Transition*
+          // series of classes removed after we successfully transition to PoS
+          // https://github.com/hyperledger/besu/issues/2897
```

Comment:
Sadly I don't think this is ever going away.  There seems to be an appetite for having merge functionality on testnets